app/server/lib/SamlConfig.ts

```diff
@@ -181,9 +190,15 @@ export class SamlConfig {
         // An example IdP response is at https://github.com/Clever/saml2#assert_response. Saml2-js
         // maps some standard attributes as user.given_name, user.surname, which we use if
         // available. Otherwise we use user.attributes which has the form {Name: [Value]}.
-        const fname = samlUser.given_name || samlUser.attributes.FirstName || '';
-        const lname = samlUser.surname || samlUser.attributes.LastName || '';
-        const email = samlUser.email || samlUser.name_id;
+        const fnameAttribute = process.env.GRIST_SAML_ATTR_FIRSTNAME || '';
+        const lnameAttribute = process.env.GRIST_SAML_ATTR_LASTNAME || '';
+        const emailAttribute = process.env.GRIST_SAML_ATTR_EMAIL || '';
+        const fname = samlUser.attributes[fnameAttribute] ||
+          samlUser.given_name || samlUser.attributes.FirstName || '';
```

Comment:
Related question: if you set a variable like `GRIST_SAML_ATTR_FIRSTNAME`, would you expect the _just_ that attribute would get used? If that attribute is empty, would you want it to fall back to looking at other attributes, or just use the empty value?

Reply:
I guess in theory "do not fallback" would make sense if the value is returned and returned empty, but I can't think of a case where the fallback values would also exist and differ